libs/logic-apps-shared/src/designer-client-services/lib/base/template.ts

```diff
@@ -0,0 +1,52 @@
+import { ArgumentException } from '../../../utils/src';
+import type { ITemplateService } from '../template';
+import type { IHttpClient } from '../httpClient';
+
+export interface TemplateServiceOptions {
+  baseUrl: string;
+  appId?: string;
+  httpClient: IHttpClient;
+  apiVersions: {
+    subscription: string;
+    gateway: string;
+  };
+  openBladeAfterCreate: () => void;
+}
+
+export class BaseTemplateService implements ITemplateService {
```

Comment:
this should be standard template service because getting existing workflow names is goingt o be different for standard and consumption